mixins/jahia.yml

```
@@ -1316,6 +1387,129 @@ actions:
+          http://localhost:8080/modules/dbconn/elasticsearch/add
+        )
+        success=$(echo $result | jq -r .success)
+        verified=$(echo $result | jq -r .connectionVerified)
+        if [ "$success" != "Connection successfully added" ]; then
+          echo "Failed to add the connection"
+          exit 1
+        elif [ "$verified" != "true" ]; then
+          echo "The new connection could not be verified"
+          exit 1
+        fi
+
+  setAugSearchESConnection:
+    - getAugSearchConnectionName
+    - if ("${globals.augSearchConnectionName}" != "jahia-cloud_augmented-search"):
+        cmd[proc]: |-
+          __secret__API_TOKEN="${globals.__secret__pat_token}"
+
+          es_connection_name="jahia-cloud_augmented-search"
+
+          response=$(curl -fLSs -XPOST \
+            -H "Authorization: APIToken $__secret__API_TOKEN" \
+            -H 'Origin: http://localhost:8080' \
+            -H 'Content-Type: application/json' \
+            -d '{"query":"mutation { admin { search { setDbConnection(connectionId:\"jahia-cloud_augmented-search\") } } }"}' \
+            http://localhost:8080/modules/graphql)
+            result=$(echo $response | jq -r '.data.admin.search.setDbConnection')
+          if [ "$result" != "Successful" ]; then
+            echo "Failed to use the $es_connection_name connection for AS: $response"
+            exit 1
+          fi
+
+  purgeASIndicesInES:
+    # Set the jahia-cloud_augmented-search connexion from elasticsearch-connector module
+    - getPapiInfoAll
+    - cmd[proc]: |-
+        __secret__PAPI_TOKEN="${globals.papiToken}"
+        export PAPI_TOKEN="$__secret__PAPI_TOKEN"
+        export PAPI_HOSTNAME="${globals.papiHostname}"
+        export PAPI_ENV_ID="${globals.papiEnvId}"
+        export PAPI_API_VERSION="${globals.papiApiVersion}"
+
+        environment=$(papi.py -X GET "paas-environment/$PAPI_ENV_ID")
+        ec_deployment_id=$(echo $environment | jq -r .ec_deployment_id)
+        ec_deployment=$(papi.py -X GET "ec-deployment/$ec_deployment_id")
+
+        ec_deployment_credentials=$(papi.py -X GET "ec-deployment/$ec_deployment_id/credentials")
+        es_endpoint=$(echo $ec_deployment | jq -r .es_endpoint)
+
+        es_password=$(echo $ec_deployment_credentials | jq -r .password)
+        es_login=$(echo $ec_deployment_credentials | jq -r .login)
+
+        indices=$(curl "$es_endpoint/_cat/indices?format=json" -u "$es_login:$es_password")
+        indices_to_delete=($(echo $indices | jq --arg pattern "$jahia_cfg_elasticsearch_prefix_." -r '.[] | select(.index|test($pattern)) | .index'))
+        for index_to_delete in ${indices_to_delete[@]}; do
+          curl -X DELETE "$es_endpoint/$index_to_delete" -u "$es_login:$es_password"
+        done
```

```suggestion
        curl -u "$es_login:$es_password" -XDELETE "$es_endpoint/${jahia_cfg_elasticsearch_prefix}_*"
```
Should do the trick (perhaps check that `jahia_cfg_elasticsearch_prefix` has the expected value as long as you do the delete with the elastic account).

At worst leave it like that but still fix the `jq --arg pattern "$jahia_cfg_elasticsearch_prefix_."` otherwise it will delete indices not related :trollface:

---

I added a check and fixed the jq pattern arg

---

I did not use ${jahia_cfg_elasticsearch_prefix} because it's interpreted by jelastic